converter/internal/test_common/testing.go

```diff
@@ -0,0 +1,138 @@
+package test_common
+
+import (
+	"bytes"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/grafana/agent/converter/diag"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	flowSuffix  = ".river"
+	diagsSuffix = ".diags"
+)
+
+// TestDirectory will execute tests for converting from a source configuration
```

Comment:
This felt warranted due to how much is wrapped up in this test, but if it feels redundant now that the function itself is much more readable I can nuke it.